backend/benefit/applications/api/v1/serializers.py

```diff
@@ -350,25 +447,42 @@ def _validate_de_minimis_aid_set(self, company, de_minimis_aid, de_minimis_aid_s
                 {"de_minimis_aid_set": "Total amount of de minimis aid too large"}
             )
 
-    def _validate_date_range(self, start_date, end_date):
+    def _validate_date_range(self, start_date, end_date, benefit_type):
         # keeping all start/end date validation together
         if start_date and start_date < date(date.today().year, 1, 1):
             raise serializers.ValidationError(
                 {"start_date": _("start_date must be within the current year")}
             )
         if end_date:
-            if start_date and end_date < start_date:
-                raise serializers.ValidationError(
-                    {
-                        "end_date": _(
-                            "application end_date can not be less than start_date"
-                        )
-                    }
-                )
             if end_date < date(date.today().year, 1, 1):
                 raise serializers.ValidationError(
                     {"end_date": _("end_date must be within the current year")}
                 )
+            if start_date:
+                if end_date < start_date:
+                    raise serializers.ValidationError(
+                        {
+                            "end_date": _(
+                                "application end_date can not be less than start_date"
+                            )
+                        }
+                    )
+                if (
+                    benefit_type != BenefitType.COMMISSION_BENEFIT
+                    and start_date + relativedelta(months=1) - relativedelta(days=1)
+                    > end_date
+                ):
+                    # A commission can have very short duration and doesn't have the 1 month minimum period as the
+                    # employment and salary based benefits.
+                    # These two option have identical duration periods which need to be between 1-12 months.
+                    # (note: we'll allow full month ranges, like 2021-02-01 - 2021-02-28
+                    raise serializers.ValidationError(
+                        {"end_date": _("minimum duration of the benefit is one month")}
+                    )
+                if start_date + relativedelta(months=12) < end_date:
+                    raise serializers.ValidationError(
+                        {"end_date": _("minimum duration of the benefit is 12 months")}
+                    )
```

Comment:
I don't know the business logic here but based on
```
# These two option have identical duration periods which need to be between 1-12 months.
```

Should this be:
```
if start_date + relativedelta(months=12) > end_date:
    raise serializers.ValidationError(
        {"end_date": _("maximum duration of the benefit is 12 months")}
    )
```

Reply:
Fixed the code and added tests
I'll be adding more validation logic in a future PR, this PR is needed in order to have something usable for the UI devs